examples/container-service-extension-3.1.x/3.7-cse-3.1.x-install.tf

```diff
@@ -230,6 +230,110 @@ resource "vcd_catalog" "cat-cse" {
   delete_recursive = "true"
 }
 
+# Create a role at provider level to be used to manage CSE (recommended).
+# You can create a `vcd_org_user` that uses this role and configure the Terraform provider to use it instead of System Administrator (recommended).
+# Here we continue to use System Administrator for simplicity and to configure CSE in one run.
```

Comment:
Hm. The last sentence is a bit confusing, i.e.: why we recommend to use Service Role, but then do not use it ourselves?

Reply:
Actually your comment gave me an idea, I'll see if it works and it could leave the HCL in a more understandable state. (I'll try to use `alias`).
https://www.terraform.io/language/providers/configuration

Reply:
Done, I refactored the HCL to use Terraform alias feature. I think it's nicer this way:
- Avoids the confusing comment and taking for granted that all customers will understand what to do.
- Preserves the "single-run" approach, just one `terraform apply` and that's it.

The idea here is that we have two provider configs, one aliased as `administrator` with Sysadmin powers, and one without alias that uses the CSE Service account. The `administrator` one creates this Service Account and some other required items (Orgs, VDCs, ALBs, etc) while the CSE Service account one creates the remaining elements.

I tested it and it worked, but I'll re-test again with a fresh VCD as I did some last-minute changes and I want to be 100% sure it works.

Reply:
I like the approach - have made some suggestions in-line. Main train of thought was: many users may not be aware of `alias` and we should try to make it really clear (in the structure of HCL and comments) that there are two providers.

> The only "disadvantage" is that the provider = vcd.administrator expression is quite repetitive, but it's a minor thing imho. We could do it the other way around and having a provider = vcd.cse-service-account, but I preferred to make it clear where admin grants were used.

I was having the same thoughts when reviewing. The question is which is "more default" in this use case? Does the configuration work without administrator rights at all? If not, it feels like administrator should be the default, while the service account - an option, esp. given that we create the service account. But these are just thoughts for discussion. 

One way to weigh the decision is to answer the question: do you have more resources requiring vcd.administrator provider or the service account provider? I.e. which would lead to less references.

Reply:
Switched to having the System Admin as the `default` alias, and the CSE Service Account as the `cse-service-account` alias. Feels less cluttered now.

Reply:
It's perfect now :) Thanks for this improvement!

Reply:
I'm resolving this conversation, but tagging @Didainius , so he's aware of this change.